### PR TITLE
Enable `Performance/UnfreezeString` cop

### PR DESCRIPTION
**Pull request by y-yagi**
In Ruby 2.3 or later, `String#+@` is available and `+@` is faster than `dup`.

```ruby
# frozen_string_literal: true

require "bundler/inline"

gemfile(true) do
  source "https://rubygems.org"

  gem "benchmark-ips"
end

Benchmark.ips do |x|
  x.report('+@') { +"" }
  x.report('dup') { "".dup }
  x.compare!
end
```

```
$ ruby -v benchmark.rb
ruby 2.5.1p57 (2018-03-29 revision 63029) [x86_64-linux]
Warming up --------------------------------------
                  +@   282.289k i/100ms
                 dup   187.638k i/100ms
Calculating -------------------------------------
                  +@      6.775M (± 3.6%) i/s -     33.875M in   5.006253s
                 dup      3.320M (± 2.2%) i/s -     16.700M in   5.032125s

Comparison:
                  +@:  6775299.3 i/s
                 dup:  3320400.7 i/s - 2.04x  slower

```

### CODE REVIEW

File: activerecord/lib/active_record/migration.rb [hunk 1]
```diff
@@ -170,7 +170,7 @@ def initialize(env = "production")
 
   class EnvironmentMismatchError < ActiveRecordError
     def initialize(current: nil, stored: nil)
-      msg =  "You are attempting to modify a database that was last run in `#{ stored }` environment.\n".dup
+      msg =  +"You are attempting to modify a database that was last run in `#{ stored }` environment.\n"
```
Comment (kamipo) [inline activerecord/lib/active_record/migration.rb]: ✂️ a space.

Comment (y-yagi) [reply activerecord/lib/active_record/migration.rb]: :+1: Fixed!

**Review by rafaelfranca — APPROVED** [review]

**Comment by matthewd** [comment]
I think this goes with #32031, which we held off on to ease backports. 

I haven't been paying enough attention to backports recently 😔 to have an opinion on whether that time has come.

**Comment by matthewd** [comment]
Extra thought: is there a cop we can use in 5-2-stable to disallow this syntax? Accidentally backporting a missing `.freeze` from #32031 would [only] harm performance on 2.2, whereas an accidental backport here would be an error (though hopefully caught by the test suite).

**Comment by y-yagi** [comment]
I agree that this will be an obstacle to the backport. 
So, for a while, I'd like to wait to merge this until 5.2.1 is released. I think that it is sufficient if 5.2.1 is released. What about it? 

> is there a cop we can use in 5-2-stable to disallow this syntax? 

Unfortunately, there seems to be no cop for that. 
We added a test that was missing when dealing with `dup`, so I think that it can probably be checked by testing. But I understand that there is concern. Add a custom cop?

**Pull request merged**